### PR TITLE
FEAT: Create the multi perceptron

### PR DESCRIPTION
**Pull request by mathieurobert1**
This pull request introduces a new `multiNeuron` class and its associated methods in `src/analyser/multi_perceptron.py`, as well as a `Perceptron` class in `src/analyser/perceptron.py`. These changes are aimed at implementing a multi-layer perceptron neural network.

### Implementation of Multi-layer Perceptron:

* [`src/analyser/multi_perceptron.py`](diffhunk://#diff-1fc92f0a8e45e5bc4910a01d7e320ca55a24119d8bc457b4fbe4c29d36daa49eR1-R78): Added the `multiNeuron` class with methods for initializing the network, making predictions, training the network, and updating parameters.
* [`src/analyser/perceptron.py`](diffhunk://#diff-e5f04f7effbec4bbf602599d4773697330695f07d901fddf190dcff670c85ee8R1-R30): Added the `Perceptron` class with methods for initialization and prediction using a sigmoid activation function.

These changes collectively enable the creation, training, and prediction capabilities of a multi-layer perceptron model.

### CODE REVIEW

**Review by theophile-jr — DISMISSED** [review]

File: src/analyser/multi_perceptron.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        for layer in self.neural_network:
+            last_inputs = []
+            for neuron in layer:
+                last_inputs.append(neuron.predict(input))
+            input = last_inputs
+        return input[0]
+
+    def train(self, inputs : list[list[int]], targets : list[int], max_iteration=10000) -> None:
+        for iteration in range(max_iteration):
+            print("\r", iteration, end="")
```
Comment (Njord201) [inline src/analyser/multi_perceptron.py:40]: Remove this debug

**Review by Njord201 — CHANGES_REQUESTED** [review]
Good work, just one debug to remove

File: src/analyser/multi_perceptron.py (hunk 1 above)
Comment (mathieurobert1) [reply src/analyser/multi_perceptron.py:40]: No this debug is good for the moment

**Review by Njord201 — APPROVED** [review]
Nice

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**